src/Rule/Nette/PresenterInjectedPropertiesExtension.php

```diff
@@ -0,0 +1,28 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rule\Nette;
+
+use PHPStan\Reflection\PropertyReflection;
+use PHPStan\Rules\Properties\ReadWritePropertiesExtension;
+
+class PresenterInjectedPropertiesExtension implements ReadWritePropertiesExtension
+{
+
+	public function isAlwaysRead(PropertyReflection $property, string $propertyName): bool
+	{
+		return false;
+	}
+
+	public function isAlwaysWritten(PropertyReflection $property, string $propertyName): bool
+	{
+		return false;
+	}
+
+	public function isInitialized(PropertyReflection $property, string $propertyName): bool
+	{
+		return $property->isPublic() &&
+			$property->getDeclaringClass()->implementsInterface('Nette\Application\IPresenter') &&
```

Comment:
This isn't just for presenters. You can skip this condition.

Reply:
oki, wrong `@inject` could hide misconfiguration error (class not processed by DI) but to find it the whole DI config would need to be analyzed.. NetteStan :-D

Reply:
I don't think this is currently a concern, we can simply consider all properties with `@inject` being initialized, it's a better state of things than what it currently reports.

But asking for a DI container is not unprecedented - phpstan-doctrine and phpstan-symfony do something similar. We can do that in the next stage if you feel like it.